[CI/CD] build-package (deb) is broken

### Component(s)

_No response_

### Describe the issue you're reporting

See recent failures https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8065610224/job/22032135180?pr=31421, https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8065600280/job/22032028700

```
# test install
echo
podman run --name $container_name -d $image_name

Error: OCI runtime error: chmod `run/shm`: Operation not supported
podman rm -fv otelcontribcol-deb-test >/dev/null 2>&1 || true
Error: Process completed with exit code 126.
```

--- a/internal/buildscripts/packaging/fpm/test.sh
+++ b/internal/buildscripts/packaging/fpm/test.sh
@@ -38,6 +38,19 @@ podman build -t $image_name -f "$SCRIPT_DIR/$pkg_type/Dockerfile.test" "$SCRIPT_
 podman rm -fv $container_name >/dev/null 2>&1 || true
 
 # test install
+CRUN_VER='1.14.4'
+mkdir -p "${HOME}/.local/bin"
+curl -L "https://github.com/containers/crun/releases/download/${CRUN_VER}/crun-${CRUN_VER}-linux-amd64" -o "${HOME}/.local/bin/crun"
+chmod +x "${HOME}/.local/bin/crun"
+mkdir -p "${HOME}/.config/containers"
+cat << EOF > "${HOME}/.config/containers/containers.conf"
+[engine.runtimes]
+crun = [
+  "${HOME}/.local/bin/crun",
+  "/usr/bin/crun"
+]
+EOF
+
 echo
 podman run --name $container_name -d $image_name
 install_pkg $container_name "$PKG_PATH"

